File: src/main/java/eterea/migration/api/rest/model/OrderNote.java
package eterea.migration.api.rest.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.json.JsonMapper;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.List;

@Data
@Entity
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class OrderNote extends Auditable {

    @Id
    private Long orderNumberId;
    private String orderStatus;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    private OffsetDateTime orderDate;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    private OffsetDateTime paidDate;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    private OffsetDateTime completedDate;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    private OffsetDateTime modifiedDate;

    private String orderCurrency;

    @Lob
    private String customerNote;

    private String billingFirstName;
    private String billingLastName;
    private String billingFullName;
    private String billingDniPasaporte;
    private String billingAddress;
    private String billingCity;
    private String billingState;
    private String billingPostCode;
    private String billingCountry;
    private String billingEmail;
    private String billingPhone;
    private String shippingFirstName;
    private String shippingLastName;
    private String shippingFullName;
    private String shippingAddress;
    private String shippingCity;
    private String shippingState;
    private String shippingPostCode;
    private String shippingCountryFull;
    private String paymentMethodTitle;
    private String cartDiscount;
    private BigDecimal orderSubtotal;
    private BigDecimal orderSubtotalRefunded;
    private String shippingMethodTitle;
    private String orderShipping;
    private BigDecimal orderShippingRefunded;
    private BigDecimal orderTotal;
    private BigDecimal orderTotalTax;

    @Lob
    private String orderNotes;

    @Lob
    private String fullPayload;

    @OneToMany(fetch = FetchType.EAGER)
    @JoinColumn(name = "orderNumberId", insertable = false, updatable = false)
    private List<Product> products;

    @OneToOne
    @JoinColumn(name = "orderNumberId", insertable = false, updatable = false)
    private Payment payment;

    public String jsonify() {
        try {
            return JsonMapper
                    .builder()
                    .findAndAddModules()
                    .build()
                    .writerWithDefaultPrettyPrinter()
                    .writeValueAsString(this);
        } catch (JsonProcessingException e) {
            return "jsonify error " + e.getMessage();
        }
    }

}

File: src/main/kotlin/eterea/migration/api/rest/kotlin/model/OrderNote.kt
package eterea.migration.api.rest.kotlin.model

import com.fasterxml.jackson.annotation.JsonFormat
import jakarta.persistence.*
import java.math.BigDecimal
import java.time.OffsetDateTime

@Entity
@Table
data class OrderNote(

    @Id
    var orderNumberId: Long? = null,
    var orderStatus: String = "",

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var orderDate: OffsetDateTime? = null,

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var paidDate: OffsetDateTime? = null,

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var completedDate: OffsetDateTime? = null,

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var modifiedDate: OffsetDateTime? = null,

    var orderCurrency: String = "",

    @Column(columnDefinition = "TEXT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci")
    var customerNote: String = "",

    var billingFirstName: String = "",
    var billingLastName: String = "",
    var billingFullName: String = "",
    var billingDniPasaporte: String = "",
    var billingAddress: String = "",
    var billingCity: String = "",
    var billingState: String = "",
    var billingPostCode: String = "",
    var billingCountry: String = "",
    var billingEmail: String = "",
    var billingPhone: String = "",
    var shippingFirstName: String = "",
    var shippingLastName: String = "",
    var shippingFullName: String = "",
    var shippingAddress: String = "",
    var shippingCity: String = "",
    var shippingState: String = "",
    var shippingPostCode: String = "",
    var shippingCountryFull: String = "",
    var paymentMethodTitle: String = "",
    var cartDiscount: String = "",
    var orderSubtotal: BigDecimal = BigDecimal.ZERO,
    var orderSubtotalRefunded: BigDecimal = BigDecimal.ZERO,
    var shippingMethodTitle: String = "",
    var orderShipping: String = "",
    var orderShippingRefunded: BigDecimal = BigDecimal.ZERO,
    var orderTotal: BigDecimal = BigDecimal.ZERO,
    var orderTotalTax: BigDecimal = BigDecimal.ZERO,

    @Lob
    @Column(columnDefinition = "TEXT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci")
    var orderNotes: String = "",

    @OneToMany
    @JoinColumn(name = "orderNumberId", insertable = false, updatable = false)
    var products: List<Product?>? = null,

    @OneToOne
    @JoinColumn(name = "orderNumberId", insertable = false, updatable = false)
    var payment: Payment? = null

) : Auditable()
